enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,825 @@
+We will extend the `template` field of the ControlPlaneMachineSet to support Cluster API in a separate enhancement.
+These fields will be added when that enhancement is implemented.
+
+#### Admission of non-authoritative resources
+
+A webhook will be introduced that will be used for admission of all Cluster API and Machine API resources
+that are involved in the sync operation.
+
+It will be responsible for validating on updates that any update to a non-authoritative resource is only performed by the sync controller itself.
+
+There will be a small number of fields that are allowed to be updated by the user:
+* `.metadata.labels` - that are not critical to Machine API or Cluster API functionality, and are not in reserved Kubernetes or OpenShift namespaces, and do not conflict with the authoritative resource
+* `.metadata.annotations` - that are not critical to Machine API or Cluster API functionality, and are not in reserved Kubernetes or OpenShift namespaces, and do not conflict with the authoritative resource
+* `.spec.authoritativeAPI` - to allow the user to set the authoritative API on the Machine API resource even when it is not authoritative
+
+Creation of resources will be monitored by the admission control and will be allowed to proceed as long as the authoritative field is set to the correct value.
+* If a Cluster API resource already exists and a Machine API resource is created, the admission will succeed only if the authoritative field is set to `cluster.x-k8s.io`
+* If a Machine API resource already exists and a Cluster API resource is created, the admission will succeed only if the authoritative field is set to `machine.openshift.io`
+
+Deletion of resources will not be monitored by the admission controller.
+
+### Implementation Details/Notes/Constraints
+
+#### Synchronisation of resources
+
+To ensure that resources are synchronised, we will implement a new controller `machine-api-sync-controller`.
+This controller will be responsible for translation of Machine API resources to Cluster API resources, and vice-versa.
+
+This controller will be responsible for creating Cluster API versions of Machine API resources, when they do not exist.
+It will observe the authoritative `spec` field described [above][authoritative-api] to synchronise between the two resources,
+overwriting anything in the non-authoritative resource that does not match with the authoritative resource.
+
+When errors occur during synchronisation, the controller will write conditions to the Machine API resource to report the failure.
+
+On successful synchronisation, the controller will update the Machine API resource's `status` to reflect the current synchronised generation of the authoritative resource,
+and set the `status.authoritativeAPI` field to the value of the authoritative resource.
+This will allow users to keep track of, and ensure that the latest changes to the authoritative resource have been synchronised.
```

Comment:
Needs revising, status authoritative API will be set separately as part of the transition